skaha/src/main/java/org/opencadc/skaha/session/GetAction.java

```diff
@@ -176,86 +179,125 @@ private ResourceStats getResourceStats() throws Exception {
                     withCores= aCPUCores;
                 }
                 
+                int rCPUCores = rCPUCoreMap.get(nodeName);
                 coresInUse = coresInUse + rCPUCores;
                 coresAvailable = coresAvailable + aCPUCores;
-                log.debug("Node: " + nodeName + " Cores: " + rCPUCores + "/" + aCPUCores + " RAM: " + aRAM + " GB");
+                log.debug("Node: " + nodeName + " Cores: " + rCPUCores + "/" + aCPUCores + " RAM: " + aRAM + " Ki");
             }
 
-            String withRAMStr = String.valueOf(withRAM) + "Gi";
-            String maxRAMStr = String.valueOf(maxRAM) + "Gi";
+            // convert RAM unit from Ki to Gi
+            String withRAMStr = String.valueOf(withRAM/1048576) + "Gi";
+            String maxRAMStr = String.valueOf(maxRAM/1048576) + "Gi";
             return new ResourceStats(desktopCount, headlessCount, totalCount, coresInUse, coresAvailable, maxCores, withRAMStr, maxRAMStr, withCores);
         } catch (Exception e) {
             log.error(e);
             throw new IllegalStateException("failed reading k8s-resources.properties", e);
```

Comment:
I think this is the wrong error message.  Something like "failed gathering statistics" would be better.

Reply:
Good eye. Updated exception message.